### ISSUE TITLE
Python "warning: unnecessary qualification"

### ISSUE THREAD

**Issue by donpellegrino** [oxigraph/oxigraph#639]
**Describe the bug**
When running `cargo build` there are 118 warnings of 'unnecessary qualification` in the python code.

```
   Compiling oxigraph_server v0.3.19 (/home/user/projects/oxigraph/server)
warning: unnecessary qualification
   --> python/src/io.rs:164:22
    |
164 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^
    |
    = note: requested on the command line with `-W unused-qualifications`

warning: unnecessary qualification
   --> python/src/io.rs:168:17
    |
168 |     fn __next__(&mut self, py: Python<'_>) -> PyResult<Option<PyTriple>> {
    |                 ^

warning: unnecessary qualification
   --> python/src/io.rs:185:22
    |
185 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/io.rs:189:17
    |
189 |     fn __next__(&mut self, py: Python<'_>) -> PyResult<Option<PyQuad>> {
    |                 ^

warning: unnecessary qualification
  --> python/src/model.rs:80:14
   |
80 |     fn value(&self) -> &str {
   |              ^

warning: unnecessary qualification
  --> python/src/model.rs:84:16
   |
84 |     fn __str__(&self) -> String {
   |                ^

warning: unnecessary qualification
  --> python/src/model.rs:88:17
   |
88 |     fn __repr__(&self) -> String {
   |                 ^

warning: unnecessary qualification
  --> python/src/model.rs:94:17
   |
94 |     fn __hash__(&self) -> u64 {
   |                 ^

warning: unnecessary qualification
  --> python/src/model.rs:98:20
   |
98 |     fn __richcmp__(&self, other: &PyAny, op: CompareOp) -> PyResult<bool> {
   |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:114:23
    |
114 |     fn __getnewargs__(&self) -> (&str,) {
    |                       ^

warning: unnecessary qualification
   --> python/src/model.rs:119:22
    |
119 |     fn __copy__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:126:30
    |
126 |     fn __deepcopy__<'a>(slf: PyRef<'a, Self>, memo: &'_ PyAny) -> PyRef<'a, Self> {
    |                              ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:207:14
    |
207 |     fn value(&self) -> &str {
    |              ^

warning: unnecessary qualification
   --> python/src/model.rs:211:16
    |
211 |     fn __str__(&self) -> String {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:215:17
    |
215 |     fn __repr__(&self) -> String {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:221:17
    |
221 |     fn __hash__(&self) -> u64 {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:225:20
    |
225 |     fn __richcmp__(&self, other: &PyAny, op: CompareOp) -> PyResult<bool> {
    |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:241:23
    |
241 |     fn __getnewargs__(&self) -> (&str,) {
    |                       ^

warning: unnecessary qualification
   --> python/src/model.rs:246:22
    |
246 |     fn __copy__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:253:30
    |
253 |     fn __deepcopy__<'a>(slf: PyRef<'a, Self>, memo: &'_ PyAny) -> PyRef<'a, Self> {
    |                              ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:338:14
    |
338 |     fn value(&self) -> &str {
    |              ^

warning: unnecessary qualification
   --> python/src/model.rs:350:17
    |
350 |     fn language(&self) -> Option<&str> {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:364:17
    |
364 |     fn datatype(&self) -> PyNamedNode {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:368:16
    |
368 |     fn __str__(&self) -> String {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:372:17
    |
372 |     fn __repr__(&self) -> String {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:378:17
    |
378 |     fn __hash__(&self) -> u64 {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:382:20
    |
382 |     fn __richcmp__(&self, other: &PyAny, op: CompareOp) -> PyResult<bool> {
    |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:398:30
    |
398 |     fn __getnewargs_ex__<'a>(&'a self, py: Python<'a>) -> PyResult<((&'a str,), &'a PyDict)> {
    |                              ^

warning: unnecessary qualification
   --> python/src/model.rs:409:22
    |
409 |     fn __copy__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:416:30
    |
416 |     fn __deepcopy__<'a>(slf: PyRef<'a, Self>, memo: &'_ PyAny) -> PyRef<'a, Self> {
    |                              ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:447:14
    |
447 |     fn value(&self) -> &str {
    |              ^

warning: unnecessary qualification
   --> python/src/model.rs:451:16
    |
451 |     fn __str__(&self) -> &str {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:455:17
    |
455 |     fn __repr__(&self) -> &str {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:459:17
    |
459 |     fn __hash__(&self) -> u64 {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:463:20
    |
463 |     fn __richcmp__(&self, other: &PyAny, op: CompareOp) -> PyResult<bool> {
    |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:479:27
    |
479 |     fn __getnewargs__<'a>(&self, py: Python<'a>) -> &'a PyTuple {
    |                           ^

warning: unnecessary qualification
   --> python/src/model.rs:484:22
    |
484 |     fn __copy__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:491:30
    |
491 |     fn __deepcopy__<'a>(slf: PyRef<'a, Self>, memo: &'_ PyAny) -> PyRef<'a, Self> {
    |                              ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:673:16
    |
673 |     fn subject(&self) -> PySubject {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:683:18
    |
683 |     fn predicate(&self) -> PyNamedNode {
    |                  ^

warning: unnecessary qualification
   --> python/src/model.rs:693:15
    |
693 |     fn object(&self) -> PyTerm {
    |               ^

warning: unnecessary qualification
   --> python/src/model.rs:697:16
    |
697 |     fn __str__(&self) -> String {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:701:17
    |
701 |     fn __repr__(&self) -> String {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:707:17
    |
707 |     fn __hash__(&self) -> u64 {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:711:20
    |
711 |     fn __richcmp__(&self, other: &Self, op: CompareOp) -> PyResult<bool> {
    |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:715:16
    |
715 |     fn __len__(&self) -> usize {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:719:20
    |
719 |     fn __getitem__(&self, input: usize) -> PyResult<PyTerm> {
    |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:728:17
    |
728 |     fn __iter__(&self) -> TripleComponentsIter {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:740:23
    |
740 |     fn __getnewargs__(&self) -> (PySubject, PyNamedNode, PyTerm) {
    |                       ^

warning: unnecessary qualification
   --> python/src/model.rs:745:22
    |
745 |     fn __copy__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:752:30
    |
752 |     fn __deepcopy__<'a>(slf: PyRef<'a, Self>, memo: &'_ PyAny) -> PyRef<'a, Self> {
    |                              ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:871:16
    |
871 |     fn subject(&self) -> PySubject {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:881:18
    |
881 |     fn predicate(&self) -> PyNamedNode {
    |                  ^

warning: unnecessary qualification
   --> python/src/model.rs:891:15
    |
891 |     fn object(&self) -> PyTerm {
    |               ^

warning: unnecessary qualification
   --> python/src/model.rs:901:19
    |
901 |     fn graph_name(&self) -> PyGraphName {
    |                   ^

warning: unnecessary qualification
   --> python/src/model.rs:911:15
    |
911 |     fn triple(&self) -> PyTriple {
    |               ^

warning: unnecessary qualification
   --> python/src/model.rs:915:16
    |
915 |     fn __str__(&self) -> String {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:919:17
    |
919 |     fn __repr__(&self) -> String {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:933:17
    |
933 |     fn __hash__(&self) -> u64 {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:937:20
    |
937 |     fn __richcmp__(&self, other: &Self, op: CompareOp) -> PyResult<bool> {
    |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:941:16
    |
941 |     fn __len__(&self) -> usize {
    |                ^

warning: unnecessary qualification
   --> python/src/model.rs:945:20
    |
945 |     fn __getitem__(&self, input: usize, py: Python<'_>) -> PyResult<PyObject> {
    |                    ^

warning: unnecessary qualification
   --> python/src/model.rs:955:17
    |
955 |     fn __iter__(&self) -> QuadComponentsIter {
    |                 ^

warning: unnecessary qualification
   --> python/src/model.rs:972:23
    |
972 |     fn __getnewargs__(&self) -> (PySubject, PyNamedNode, PyTerm, PyGraphName) {
    |                       ^

warning: unnecessary qualification
   --> python/src/model.rs:982:22
    |
982 |     fn __copy__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/model.rs:989:30
    |
989 |     fn __deepcopy__<'a>(slf: PyRef<'a, Self>, memo: &'_ PyAny) -> PyRef<'a, Self> {
    |                              ^^^^^

warning: unnecessary qualification
    --> python/src/model.rs:1048:14
     |
1048 |     fn value(&self) -> &str {
     |              ^

warning: unnecessary qualification
    --> python/src/model.rs:1052:16
     |
1052 |     fn __str__(&self) -> String {
     |                ^

warning: unnecessary qualification
    --> python/src/model.rs:1056:17
     |
1056 |     fn __repr__(&self) -> String {
     |                 ^

warning: unnecessary qualification
    --> python/src/model.rs:1060:17
     |
1060 |     fn __hash__(&self) -> u64 {
     |                 ^

warning: unnecessary qualification
    --> python/src/model.rs:1064:20
     |
1064 |     fn __richcmp__(&self, other: &Self, op: CompareOp) -> PyResult<bool> {
     |                    ^

warning: unnecessary qualification
    --> python/src/model.rs:1069:23
     |
1069 |     fn __getnewargs__(&self) -> (&str,) {
     |                       ^

warning: unnecessary qualification
    --> python/src/model.rs:1074:22
     |
1074 |     fn __copy__(slf: PyRef<'_, Self>) -> PyRef<Self> {
     |                      ^^^^^

warning: unnecessary qualification
    --> python/src/model.rs:1081:30
     |
1081 |     fn __deepcopy__<'a>(slf: PyRef<'a, Self>, memo: &'_ PyAny) -> PyRef<'a, Self> {
     |                              ^^^^^

warning: unnecessary qualification
    --> python/src/model.rs:1351:22
     |
1351 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
     |                      ^^^^^

warning: unnecessary qualification
    --> python/src/model.rs:1355:17
     |
1355 |     fn __next__(&mut self) -> Option<PyTerm> {
     |                 ^

warning: unnecessary qualification
    --> python/src/model.rs:1367:22
     |
1367 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
     |                      ^^^^^

warning: unnecessary qualification
    --> python/src/model.rs:1371:17
     |
1371 |     fn __next__(&mut self, py: Python<'_>) -> Option<PyObject> {
     |                 ^

warning: unnecessary qualification
  --> python/src/sparql.rs:98:17
   |
98 |     fn __repr__(&self) -> String {
   |                 ^

warning: unnecessary qualification
   --> python/src/sparql.rs:111:20
    |
111 |     fn __richcmp__(&self, other: &Self, op: CompareOp) -> PyResult<bool> {
    |                    ^

warning: unnecessary qualification
   --> python/src/sparql.rs:121:16
    |
121 |     fn __len__(&self) -> usize {
    |                ^

warning: unnecessary qualification
   --> python/src/sparql.rs:125:20
    |
125 |     fn __getitem__(&self, input: &PyAny) -> PyResult<Option<PyTerm>> {
    |                    ^

warning: unnecessary qualification
   --> python/src/sparql.rs:144:17
    |
144 |     fn __iter__(&self) -> SolutionValueIter {
    |                 ^

warning: unnecessary qualification
   --> python/src/sparql.rs:158:22
    |
158 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/sparql.rs:162:17
    |
162 |     fn __next__(&mut self) -> Option<Option<PyTerm>> {
    |                 ^

warning: unnecessary qualification
   --> python/src/sparql.rs:187:18
    |
187 |     fn variables(&self) -> Vec<PyVariable> {
    |                  ^

warning: unnecessary qualification
   --> python/src/sparql.rs:195:22
    |
195 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/sparql.rs:199:17
    |
199 |     fn __next__(&mut self) -> PyResult<Option<PyQuerySolution>> {
    |                 ^

warning: unnecessary qualification
   --> python/src/sparql.rs:220:22
    |
220 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/sparql.rs:224:17
    |
224 |     fn __next__(&mut self) -> PyResult<Option<PyTriple>> {
    |                 ^

warning: unnecessary qualification
   --> python/src/store.rs:126:12
    |
126 |     fn add(&self, quad: &PyQuad, py: Python<'_>) -> PyResult<()> {
    |            ^

warning: unnecessary qualification
   --> python/src/store.rs:147:15
    |
147 |     fn extend(&self, quads: &PyAny, py: Python<'_>) -> PyResult<()> {
    |               ^

warning: unnecessary qualification
   --> python/src/store.rs:172:20
    |
172 |     fn bulk_extend(&self, quads: &PyAny) -> PyResult<()> {
    |                    ^

warning: unnecessary qualification
   --> python/src/store.rs:194:15
    |
194 |     fn remove(&self, quad: &PyQuad, py: Python<'_>) -> PyResult<()> {
    |               ^

warning: unnecessary qualification
   --> python/src/store.rs:221:9
    |
221 |         &self,
    |         ^

warning: unnecessary qualification
   --> python/src/store.rs:278:9
    |
278 |         &self,
    |         ^

warning: unnecessary qualification
   --> python/src/store.rs:333:15
    |
333 |     fn update(&self, update: &str, base_iri: Option<&str>, py: Python<'_>) -> PyResult<()> {
    |               ^

warning: unnecessary qualification
   --> python/src/store.rs:379:9
    |
379 |         &self,
    |         ^

warning: unnecessary qualification
   --> python/src/store.rs:461:9
    |
461 |         &self,
    |         ^

warning: unnecessary qualification
   --> python/src/store.rs:539:9
    |
539 |         &self,
    |         ^

warning: unnecessary qualification
   --> python/src/store.rs:591:21
    |
591 |     fn named_graphs(&self) -> GraphNameIter {
    |                     ^

warning: unnecessary qualification
   --> python/src/store.rs:608:29
    |
608 |     fn contains_named_graph(&self, graph_name: &PyAny) -> PyResult<bool> {
    |                             ^

warning: unnecessary qualification
   --> python/src/store.rs:629:18
    |
629 |     fn add_graph(&self, graph_name: &PyAny, py: Python<'_>) -> PyResult<()> {
    |                  ^

warning: unnecessary qualification
   --> python/src/store.rs:659:20
    |
659 |     fn clear_graph(&self, graph_name: &PyAny, py: Python<'_>) -> PyResult<()> {
    |                    ^

warning: unnecessary qualification
   --> python/src/store.rs:682:21
    |
682 |     fn remove_graph(&self, graph_name: &PyAny, py: Python<'_>) -> PyResult<()> {
    |                     ^

warning: unnecessary qualification
   --> python/src/store.rs:710:14
    |
710 |     fn clear(&self, py: Python<'_>) -> PyResult<()> {
    |              ^

warning: unnecessary qualification
   --> python/src/store.rs:720:14
    |
720 |     fn flush(&self, py: Python<'_>) -> PyResult<()> {
    |              ^

warning: unnecessary qualification
   --> python/src/store.rs:730:17
    |
730 |     fn optimize(&self, py: Python<'_>) -> PyResult<()> {
    |                 ^

warning: unnecessary qualification
   --> python/src/store.rs:755:15
    |
755 |     fn backup(&self, target_directory: &str, py: Python<'_>) -> PyResult<()> {
    |               ^

warning: unnecessary qualification
   --> python/src/store.rs:763:16
    |
763 |     fn __str__(&self, py: Python<'_>) -> String {
    |                ^

warning: unnecessary qualification
   --> python/src/store.rs:767:17
    |
767 |     fn __bool__(&self) -> PyResult<bool> {
    |                 ^

warning: unnecessary qualification
   --> python/src/store.rs:771:16
    |
771 |     fn __len__(&self) -> PyResult<usize> {
    |                ^

warning: unnecessary qualification
   --> python/src/store.rs:775:21
    |
775 |     fn __contains__(&self, quad: &PyQuad) -> PyResult<bool> {
    |                     ^

warning: unnecessary qualification
   --> python/src/store.rs:779:17
    |
779 |     fn __iter__(&self) -> QuadIter {
    |                 ^

warning: unnecessary qualification
   --> python/src/store.rs:793:22
    |
793 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/store.rs:797:17
    |
797 |     fn __next__(&mut self) -> PyResult<Option<PyQuad>> {
    |                 ^

warning: unnecessary qualification
   --> python/src/store.rs:812:22
    |
812 |     fn __iter__(slf: PyRef<'_, Self>) -> PyRef<Self> {
    |                      ^^^^^

warning: unnecessary qualification
   --> python/src/store.rs:816:17
    |
816 |     fn __next__(&mut self) -> PyResult<Option<PyNamedOrBlankNode>> {
    |                 ^

warning: `pyoxigraph` (lib) generated 118 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 3m 22s
```

**To Reproduce**
1. Which version of Oxigraph are you using? On which platform? - main branch as of the filing of this report, Ubuntu 22.04.3 LTS, Python 3.10.12 installed 
2. A command line or a code snippet that triggers the bug. - `cargo build`

**Workaround**

I presume these warnings are inconsequential for execution, but it might be useful to clean them up at some point so that they do not distract from warnings generated against new code changes that could be more informative.

**Comment by Tpt** [comment]
Thank you for reporting this. Indeed these warnings are inconsequential.

I have fixed this in the `next` branch (my WIP for Oxigraph 0.4 with mostly query planning and parser changes).

A quick&dirty way to hide them is to add `#![allow(unused_qualifications)]` at in the root `lib.rs` file of pyoxigraph

**Comment by Tpt** [comment]
Fixed in 0.4.0-alpha.1
